django/db/models/sql/compiler.py

```diff
@@ -150,7 +147,14 @@ def collapse_group_by(self, expressions, having):
                         getattr(expr.output_field, 'model') == self.query.model):
                     pk = expr
             if pk:
+                # MySQLism: Columns in HAVING clause must be added to the GROUP BY.
```

Comment:
It's good that you've documented that this condition is a mysql-ism, but I'm a bit suspect about including behaviour specific to mysql in a block that is gated by a feature that (theoretically) could apply to other backends. I guess we can deal with it if it becomes a problem (which this comment will help to find).